test/e2e/util.go

```diff
@@ -0,0 +1,17 @@
+package main
+
+import (
+	capiv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+// MachineByCreationTimestamp sorts a list of MachineSet by creation timestamp, using their names as a tie breaker.
+type MachineByCreationTimestamp []capiv1alpha1.Machine
+
+func (o MachineByCreationTimestamp) Len() int      { return len(o) }
+func (o MachineByCreationTimestamp) Swap(i, j int) { o[i], o[j] = o[j], o[i] }
+func (o MachineByCreationTimestamp) Less(i, j int) bool {
+	if o[i].CreationTimestamp.Equal(&o[j].CreationTimestamp) {
+		return o[i].Name < o[j].Name
+	}
+	return o[i].CreationTimestamp.After(o[j].CreationTimestamp.Time)
+}
```

Comment:
All of the swap functionality can be replaced and simplified by using `sort.Slice()`.

```go
sort.Slice(o, func(i, j int) bool {
	if o[i].CreationTimestamp.Equal(&o[j].CreationTimestamp) {
		return o[i].Name < o[j].Name
	}
	return o[i].CreationTimestamp.After(o[j].CreationTimestamp.Time)
})
```

Reply:
thanks! I didn't know that. If you are ok with it I'd like to keep the interface impl. rather than inline it since I find it more readable/reusable

Reply:
I still think you would have less code. You can still create a wrapper func:

```go
func sortByCreationTimestamp() ... 
```

with the implementation I posted. For me this is win/win; using stdlib code (so no surprises) and reusable as it is encapsulated by a named function.

Reply:
changed, thanks